src/globals.c

```diff
@@ -543,3 +545,10 @@ void free_globals(void)
 		bundle_to_add = NULL;
 	}
 }
+
+void save_cmd(int argc, char **argv)
+{
+	for (int i = 0; i < argc; i++) {
+		string_or_die(&swupd_cmd, "%s %s", swupd_cmd, argv[i]);
```

Comment:
Forgive my selective reading, you're right. The leaks do need to be fixed still, but it indeed saves the whole string. I just ignored the swupd_cmd being passed to the first formatter %s and assumed you were treating argv[1] as being --path /blah/blah. Thanks for clearing it up :)